packages/router/src/lib/helpers/prepare.ts

```diff
@@ -33,39 +35,73 @@ export const validBidExpiry = (bidExpiry: number, currentTime: number) => bidExp
  * Returns the swapRate
  *
  * @param TODO
- * @returns The swapRate, determined by the AMM
+ * @returns The swapRate, determined by the stableSwap
  *
  * @remarks
- * TODO: getSwapRate using AMM
+ * TODO: getSwapRate using stableSwap
  */
-export const getSwapRate = async (): Promise<string> => {
-  return "1";
+export const getSwapRate = async (
+  amount: BigNumber,
+  balances: BigNumber[],
+  indexIn: number,
+  indexOut: number,
+): Promise<BigNumber> => {
+  const amountOut = getAmountsOut(
```

Comment:
will the `getAmountsOut` function properly return only integers?

Reply:
absolutely!